src/app/data-planes/data/index.ts

```diff
@@ -0,0 +1,202 @@
+  lastUpdated: Date
+  version: string
+  compatible: boolean,
+  dataplane: Dataplane
+  dataplaneInsight: DataplaneInsight
+}
+
+type Label = {
+  label: string
+  value: string
+}
+type BuildVersion = KumaDpVersion | EnvoyVersion | { version: string}
+///
+
+export const Dataplane = {
+  fromJSONCollection(collection: CollectionResponse<unknown>): CollectionResponse<Dataplane> {
+    collection.items = collection.items.map((item: unknown) => {
+      return Dataplane.fromJSON(item)
+    })
+    return collection as CollectionResponse<Dataplane>
+  },
+  fromJSON(object: unknown): Dataplane {
+    const item = {
+      ...(object as OriginalDataplane),
+    } as Dataplane
+    // make sure networking.inbound is at least an empty array
+    item.networking = DataplaneNetworking.fromJSON(item.networking)
+    //
+    item.tags = getTags(item)
+    item.service = getTag(item, 'kuma.io/service')
+    item.protocol = getTag(item, 'kuma.io/protocol')
+    item.zone = getTag(item, 'kuma.io/zone')
+    return item
+  },
+}
+
+export const DataplaneInsight = {
+  fromJSON(object: unknown): DataplaneInsight {
+    const original = object as OriginalDataplaneInsight
+    const item = {
+      ...original,
+    } as DataplaneInsight
+
+    return {
+      ...item as DataplaneInsight,
+      subscriptions: Array.isArray(original?.subscriptions) ? original.subscriptions : [],
+    }
+  },
+}
+export const DataplaneOverview = {
+  fromJSONCollection(collection: CollectionResponse<unknown>): CollectionResponse<DataplaneOverview> {
+    collection.items = collection.items.map((item: unknown) => {
+      return DataplaneOverview.fromJSON(item)
+    })
+    return collection as CollectionResponse<DataplaneOverview>
+  },
+  fromJSON(object: unknown): DataplaneOverview {
+    const item = {
+      ...(object as OriginalDataplaneOverview),
+    } as DataplaneOverview
+    // make sure dataplane.networking.inbound is at least an empty array
+    item.dataplane = Dataplane.fromJSON(item.dataplane)
+    // make sure subscriptions is at least an empty array
+    item.dataplaneInsight = DataplaneInsight.fromJSON(item.dataplaneInsight || {})
+    //
+    item.status = getStatus(item)
+    const dependencies = getDependencies(item.dataplaneInsight)
+    item.lastUpdated = getLastUpdated(item.dataplaneInsight)
+    item.version = dependencies.kumaDp.version
+    const compatibility = compatibilityKind(
+      {
+        kumaDp: dependencies.kumaDp as KumaDpVersion,
+        envoy: dependencies.envoy as EnvoyVersion,
+        dependencies: {} as Record<string, string>,
+      },
+    ).kind
+    item.compatible = compatibility !== COMPATIBLE
+    return item
+  },
+}
+
+// utilities
+function getStatus(data: DataplaneOverview): string {
+  const dataplane = data.dataplane
+  const dataplaneInsight = data.dataplaneInsight
+  const errors = dataplane.networking.inbound
+    .filter(item => {
+      // @TODO
+      return item.health && !item.health.ready
```

Comment:
@lahabana from what I remember from the `health` property. It looks like if `health` isn't set at all that essentially means that it is exempt from the "status/health" check that this function does. Following on from that it sounds/looks/feels like that if health isn't set we assume the dataplane is healthy? Did I follow that correctly?

Reply:
Ah ok, is this potentially related to the "information not available" thing from external services? I could do with chatting through this with someone.

Reply:
Just to note: I chatted with @lahabana about this and cleared up my questions, ty! Resolving this one